feat: AWSNodeTemplate Static Drift (#4345)
aws/karpenter: website/content/en/preview/upgrade-guide.md

## Released Upgrade Notes

### Upgrading to v0.30.0+

> * Karpenter will now statically drift on both Provisioner and AWSNodeTemplate Fields. For Provisioner Static Drift, the annotation `karpenter.sh/provisioner-hash` must be present on both the Provisioner and Machine. For AWSNodeTemplate drift, the annotation `karpenter.k8s.aws/nodetemplate-hash` annotation must be present on the AWSNodeTemplate and Machine. Karpenter will not add these annotations to pre-existing nodes so these nodes will need to be recycled one time for the annotations to be added.

Reviewer: Should you link to the section that define static drift here? Without knowing what that actually means, it's unclear to a user what we are drifting on.

> * Karpenter will now statically drift on both Provisioner and AWSNodeTemplate Fields. For Provisioner Static Drift, the annotation `karpenter.sh/provisioner-hash` must be present on both the Provisioner and Machine. For AWSNodeTemplate drift, the annotation `karpenter.k8s.aws/nodetemplate-hash` annotation must be present on the AWSNodeTemplate and Machine. Karpenter will not add these annotations to pre-existing nodes so these nodes will need to be recycled one time for the annotations to be added.

Reviewer: ```suggestion
* Karpenter will now statically drift on both Provisioner and AWSNodeTemplate Fields. For Provisioner Static Drift, the `karpenter.sh/provisioner-hash` annotation must be present on both the Provisioner and Machine. For AWSNodeTemplate drift, the `karpenter.k8s.aws/nodetemplate-hash` annotation must be present on the AWSNodeTemplate and Machine. Karpenter will not add these annotations to pre-existing nodes, so each of these nodes will need to be recycled one time for the annotations to be added.
```

> * karpenter will now statically drift on both provisioner and AWSNodeTemplate Fields. For Provisioner Static Drift, the annotation `karpenter.sh/provisioner-hash` must be present on both the Provisioner and Machine. For AWSNodeTemplate drift, the annotation `karpenter.k8s.aws/nodetemplate-hash` annotation must be present on the AWSNodeTemplate and Machine. Karpenter will add the annotation for the Provisioner and AWSNodeTemplate, however for pre-existing nodes that are manged by karpenter will need to be recycled one time for them to contain the annotations. This will be for karpenter to verify the state of the nodes relative to the owning CRDs. 

Reviewer: ```suggestion
* Karpenter will now statically drift on both Provisioner and AWSNodeTemplate Fields. For Provisioner Static Drift, the annotation `karpenter.sh/provisioner-hash` must be present on both the Provisioner and Machine. For AWSNodeTemplate drift, the annotation `karpenter.k8s.aws/nodetemplate-hash` annotation must be present on the AWSNodeTemplate and Machine. Karpenter will not add these annotations to pre-existing nodes so these nodes will need to be recycled one time for the annotations to be added.
```
